internal/app/server/handlers/handler.go

```diff
@@ -246,15 +250,22 @@ func (h *Handler) deleteShorURLs(w http.ResponseWriter, r *http.Request) {
 
 }
 
+// CloseDeleteChannel Завершение чтения задач на удаление ссылок
+func (h *Handler) CloseDeleteChannel() {
+	close(h.deleteJobs)
```

Comment:
Если просто взять и закрыть канал, то все горутины, запущенные из deleteShorURLs и не успевшие закончить работу, попытаются писать в закрытый канал и приведут к панике. В рамках уже сделанного имеет смысл добавить еще одну WaitGroup внутри структуры Handler и в запускаемой горутине выполнять

```
	h.delWG.Add(1)
	go func() {
		defer h.delWG.Done()
		h.deleteJobs <- deleteJob{
			shortURL: shortURLs,
			userID:   userID,
		}
	}()
```
а в данной функции перед закрытием канала:
```suggestion
	h.delWG.Wait()
	close(h.deleteJobs)
```
В этом случае мы сначала дождемся завершения всех горутин, потом закроем канал, который приведет завершению deleteWorker, сбросу wg на уровне mainи аккуратному завершению всех процессов.